Python/Product/PythonTools/PythonTools/Commands/ImportCoverageCommand.cs

```diff
@@ -0,0 +1,126 @@
+// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.PythonTools.CodeCoverage;
+using Microsoft.PythonTools.Parsing;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudioTools;
+
+namespace Microsoft.PythonTools.Commands {
+    /// <summary>
+    /// Provides the command for importing code coverage information from a coverage.py XML file.
+    /// </summary>
+    internal sealed class ImportCoverageCommand : Command {
+        private readonly IServiceProvider _serviceProvider;
+
+        public ImportCoverageCommand(IServiceProvider serviceProvider) {
+            _serviceProvider = serviceProvider;
+            
+        }
+
+        public override void DoCommand(object sender, EventArgs args) {
+            var oe = args as OleMenuCmdEventArgs;
+            string file = oe.InValue as string;
+            PythonLanguageVersion? version = null;
+            if (file == null) {
+                object[] inp = oe.InValue as object[];
+                if (inp != null && inp.Length == 2) {
+                    file = inp[0] as string;
+                    if (inp[1] is PythonLanguageVersion) {
+                        version = (PythonLanguageVersion)inp[1];
+                    }
+                }
+            }
+
+            if (file == null) {
+                file = _serviceProvider.BrowseForFileOpen(
+                    IntPtr.Zero,
+                    "Coverage XML (*.xml)|*.xml|All Files (*.*)|*.*"
+                );
+            }
+
+            if (file != null) {
+                var outFilename = Path.Combine(
+                    Path.GetDirectoryName(file), 
```

Comment:
Which calls are you suggesting should be replaced w/ PathUtils calls?  PathUtils doesn't have Combine, GetDirectoryName, or GetFileNameWithoutExtension.  And long paths don't really matter here because we're not going to be able to write to them anyway.

---
In reply to: [67383829](https://github.com/Microsoft/PTVS/pull/1337#discussion_r67383829) [](ancestors = 67383829)